### ISSUE TITLE
A/libc: Fatal signal 11 (SIGSEGV), code 1, fault addr 0x69 in tid 3715

### ISSUE THREAD

**Issue by Queatz** [realm/realm-java#3294]
#### Goal

Clearer error message.

I had accidentally auto-completed a `getString` as a `getShort` somewhere in my codebase.  Took a bit of time to track it down due to mystical error message.

This is using a `DynamicRealm`.

#### Expected Results

Error that says something like `Type error: cannot convert string field to short`

#### Actual Results

Get this single line error:
```
A/libc: Fatal signal 11 (SIGSEGV), code 1, fault addr 0x69 in tid 3715
```

#### Steps & Code to Reproduce

Call `getShort("field")` on a field defined as `schema.addField("field", String.class)`.

#### Code Sample

```java

> Your code here. Bigger samples should ideally be as separate Android Studio project, 
> in gists/repositories or privately at help@realm.io)

```

#### Version of Realm and tooling
Realm version(s): **1.1.1**

Android Studio version: **2.2 Beta**

Which Android version and device: **Android 5.1** on a Huawei Mate S

**Comment by cmelchior** [comment]
Thanks for reporting this, and your right. It should show a better exception message.

**Comment by cmelchior** [comment]
Closed by #3312
